libs/shared/flagd-core/src/lib/parser.spec.ts

```diff
@@ -1,75 +1,117 @@
 
-  it('should fail if invalid - missing default value', () => {
-    const invalidFlag =
-      '{\n' +
-      '  "flags": {\n' +
-      '    "myBoolFlag": {\n' +
-      '      "state": "ENABLED",\n' +
-      '      "variants": {\n' +
-      '        "on": true,\n' +
-      '        "off": false\n' +
-      '      }\n' +
-      '    }\n' +
-      '  }\n' +
-      '}';
-
-    expect(() => parse(invalidFlag)).toThrowError();
-  });
+      const flags = parse(longFlag, false);
+      expect(flags).toBeTruthy();
+      expect(flags.get('myBoolFlag')).toBeTruthy();
+      expect(flags.get('myStringFlag')).toBeTruthy();
+    });
 
-  it('should parse flag configurations with references', () => {
-    const flagWithRef =
-      '{"flags":{"fibAlgo":{"variants":{"recursive":"recursive","binet":"binet"},"defaultVariant":"recursive","state":"ENABLED","targeting":{"if":[{"$ref":"emailSuffixFaas"},"binet",null]}}},"$evaluators":{"emailSuffixFaas":{"in":["@faas.com",{"var":["email"]}]}}}';
+    it('should fail if invalid - missing default value', () => {
+      const invalidFlag =
+        '{\n' +
+        '  "flags": {\n' +
+        '    "myBoolFlag": {\n' +
+        '      "state": "ENABLED",\n' +
+        '      "variants": {\n' +
+        '        "on": true,\n' +
+        '        "off": false\n' +
+        '      }\n' +
+        '    }\n' +
+        '  }\n' +
+        '}';
 
-    const flags = parse(flagWithRef);
-    expect(flags).toBeTruthy();
+      expect(() => parse(invalidFlag, false)).toThrowError();
+    });
 
-    const fibAlgo = flags.get('fibAlgo');
-    expect(fibAlgo).toBeTruthy();
-    expect(fibAlgo?.targeting).toStrictEqual({ if: [{ in: ['@faas.com', { var: ['email'] }] }, 'binet', null] });
-  });
+    it('should parse flag configurations with references', () => {
+      const flagWithRef =
+        '{"flags":{"fibAlgo":{"variants":{"recursive":"recursive","binet":"binet"},"defaultVariant":"recursive","state":"ENABLED","targeting":{"if":[{"$ref":"emailSuffixFaas"},"binet",null]}}},"$evaluators":{"emailSuffixFaas":{"in":["@faas.com",{"var":["email"]}]}}}';
+
+      const flags = parse(flagWithRef, false);
+      expect(flags).toBeTruthy();
+
+      const fibAlgo = flags.get('fibAlgo');
+      expect(fibAlgo).toBeTruthy();
+      expect(fibAlgo?.targeting).toStrictEqual({ if: [{ in: ['@faas.com', { var: ['email'] }] }, 'binet', null] });
+    });
+
+    it('should throw a parsing error due to invalid JSON', () => {
+      const invalidJson = '{';
+
+      expect(() => parse(invalidJson, false)).toThrow(ParseError);
+    });
+
+    it('should throw a parsing error due to invalid flagd configuration', () => {
+      const invalidFlagdConfig = '{"flags":{"fibAlgo":{}}}';
+
+      expect(() => parse(invalidFlagdConfig, false)).toThrow(ParseError);
```

Comment:
None of these tests have changed, I just put them in a new `describe` block.

The last 2 tests in this file were added.